### ISSUE TITLE
CfP submission form loses data when creating an account or logging in

### ISSUE THREAD

**Issue by glyph** [pretalx/pretalx#1742]
### Problem and impact

As a submitter to CfPs using pretalx, I frequently lose data while filling out the form.  This can be quite frustrating as some CfPs have multiple multi-hundred-word questions and I can type for quite a while.

### Expected behaviour

I would expect that any input I start putting into the form will be auto-saved, particularly if I have started filling out the form without being authenticated, then log in or register.

### Steps to reproduce

1. Go to a conference running a pretalx CfP
2. Start filling out the form
3. See that I'm not logged in
4. Go to the "register" link
5. Fill out email address & password
6. Come back to see the site blanked out.

**Comment by rixx** [comment]
Hm, yeah, that sucks. Two questions to figure out how to proceed:

1. Could you expand on the critical step, 'Go to the "register" link'? Do you mean that you click the "login" link in the upper right-hand corner, or that you follow the CfP (multi-step) form and eventually are asked to log in or register?
The reason I'm asking is that these are two very distinct bug reports – we extensively test for the latter, as that's the expected workflow. It's meant to give users a (seamless-ish?) way to log in or register while in the process of filling in their proposal information. 
2. What do you mean by "Come back to see the site blanked out." – by "blanked out" you mean empty form fields, not a white page, right? Did you open the login/registration form in a separate tab and switched back to the previous one? On page reload, are you thrown back to the first form of the CfP? Or did you log in then navigate back in the browser (which I could see leading to CSRF errors, hm).

Things I just tested manually:

- Submit first form, be directed to login/register page, log in, continue: works.
- Submit first form, be directed to login/register page, open other login page in separate tab, log in, reload page. See 404 (not ideal!), but on using browser navs to go back a step to the first form and hard-reloading, all info is still there.
- Start filling in first form, do **not** submit, log in in separate tab, then submit form in first tab: Get CSRF error. Navigating back with browser navigation shows me the previously entered data, and soft refresh retains the data, but that's browser dependent behaviour – Firefox keeps form data across soft refresh, Chrome doesn't.
- Start filling in first form, do **not** submit, log in in separate tab, refresh: data is lost or not as above depending on browser behaviour.

I feel like forms being emptied of data when reloading without submitting is right on the edge of "expected, correct behaviour, do not mess with it" and "annoying af, do magic with localstorage to prevent". (Any data loss following a form submit is *definitely* a bug and needs to be treated as such of course.) I do think we need to err on the side of messing with JavaScript and LocalStorage, as much as it pains me, because this isn't the first time I've heard people complain of this, and being useful is more important than being correct etc etc.

**Comment by glyph** [comment]
> 1. Could you expand on the critical step, 'Go to the "register" link'? Do you mean that you click the "login" link in the upper right-hand corner, or that you follow the CfP (multi-step) form and eventually are asked to log in or register?

The login link in the upper right hand corner is the one I just experienced.

**Comment by glyph** [comment]
> 2\. What do you mean by "Come back to see the site blanked out." – by "blanked out" you mean empty form fields, not a white page, right? Did you open the login/registration form in a separate tab and switched back to the previous one? On page reload, are you thrown back to the first form of the CfP? Or did you log in then navigate back in the browser (which I could see leading to CSRF errors, hm).

Yes, empty form fields. I did not open a new tab; just clicked the link, then ended up on the blank form, hit "back" to see if the state was saved in my page history, and it still wasn't, that one was blank as well.

**Comment by glyph** [comment]
> I feel like forms being emptied of data when reloading without submitting is right on the edge of "expected, correct behaviour, do not mess with it" and "annoying af, do magic with localstorage to prevent". (Any data loss following a form submit is _definitely_ a bug and needs to be treated as such of course.) I do think we need to err on the side of messing with JavaScript and LocalStorage, as much as it pains me, because this isn't the first time I've heard people complain of this, and being useful is more important than being correct etc etc.

Thanks, yeah, I think the LocalStorage option is the one I'd like to see.

(Is PyScript at the point where you could avoid JavaScript, I wonder? Or would that destroy page load performance? :))

**Comment by glyph** [comment]
A quick fix to avoid this frustration might be to add a speedbump on *any* navigation away from the page, "are you sure you don't want to save first", but a nice background autosave would be preferable.

**Comment by rixx** [comment]
> A quick fix to avoid this frustration might be to add a speedbump on any navigation away from the page, "are you sure you don't want to save first"

This exists in pretalx as of the last release, as per #1107 – the instance we're (probably? nbpy?) talking about is lagging by a year / two big releases atm, not much I can do there, sadly.

**Comment by glyph** [comment]
> the instance we're (probably? nbpy?) talking about

good guess :)

**Comment by glyph** [comment]
To be fair I have experienced this problem on PyCon several times as well. Glad to know that the speedbump exists.
